app/src/test/java/com/github/se/eventradar/ChatViewModelUnitTest.kt

```diff
@@ -270,4 +272,60 @@ class ChatViewModelUnitTest {
     assert(uiState.messageBarInput == "")
     assert(uiState.messageHistory.messages.isEmpty())
   }
+
+  @Test
+  fun `observeMessages Success`() = runTest {
+    (userRepository as MockUserRepository).updateCurrentUserId("user1")
+    userRepository.addUser(opponent)
+
+    val msg1 = mockMessage.copy(sender = "user1", id = "msg1")
+    val msg2 = mockMessage.copy(sender = "user2", id = "msg2")
+
+    val messageHistory = messageRepository.createNewMessageHistory("user1", "user2")
+
+    val editedMH = (messageHistory as Resource.Success).data
+
+    editedMH.messages.add(msg1)
+
+    // Init calls observeMessages()
+    viewModel = ChatViewModel(messageRepository, userRepository, opponentId)
+
+    messageRepository.addMessage(msg2, editedMH)
+
+    val expected = messageRepository.getMessages("user1", "user2")
+
+    (messageRepository as MockMessageRepository).messagesFlow.emit(expected)
+
+    assert(viewModel.uiState.value.messageHistory == (expected as Resource.Success).data)
+  }
+
+  @Test
+  fun `observeMessages Failure`() = runTest {
+    mockkStatic(Log::class)
+    every { Log.d(any(), any()) } returns 0
+    (userRepository as MockUserRepository).updateCurrentUserId("user1")
+    userRepository.addUser(opponent)
+
+    val exception = Exception("Error retrieving message histories")
+    (messageRepository as MockMessageRepository).messagesFlow.emit(Resource.Failure(exception))
+
+    viewModel = ChatViewModel(messageRepository, userRepository, opponentId)
+
+    delay(500)
+    val expectedLogMessage = "Failed to fetch messages: ${exception.message}"
+
+    val expectedMH =
```

Comment:
These will likely need to be updated after the message repo update, will communicate offline for help with this